### DISCUSSION TITLE
Why does emqx_auth_redis ACL query command use hgetall?

### DISCUSSION THREAD

**Discussion by antkai** [emqx/emqx#8021] [General]
### Subject of the support
This is my first time to use emqx. 
When I learned about using redis as an ACL plug-in, 
I saw such a configuration in the configuration file:
```conf
## ACL query command.
##
## Value: Redis cmd
##
## Variables:
##  - %u: username
##  - %c: clientid
auth.redis.acl_cmd = HGETALL mqtt_acl:%u
```
I wonder why "HGET mqtt_acl:%u %t" is not used. 
The performance of "hget" should be better than that of "hgetall".
After testing, I found that the placeholder `%t` here is not supported. In other commands, `%t` is used to refer to topic.
I hope you can answer my doubts, thanks.

### Your environment
- EMQ X version: EMQX Community Edition v4.4.3
- Plugin: emqx_auth_redis
- GitHub: https://github.com/emqx/emqx/tree/v4.4.3/apps/emqx_auth_redis

**Comment by zmstone** [comment]
If you are sure there’s only one value per user, you can change it to HGET.
(I didn’t test it though)

**Comment by zmstone** [comment]
As for place holders, data in external backends backed are modeled around the clients, but not topics.
Also, the alc cache is per client.
I.e. whenever a client subscribes, it needs to know the full picture of all rules for this client.
